dcos-log/journal/reader/read_test.go

```diff
@@ -195,3 +198,71 @@ func TestOptionMatchOR(t *testing.T) {
 		t.Fatalf("Expecting to find 2 entries, got %d", size)
 	}
 }
+
+func TestFollow(t *testing.T) {
+	id := getUniqueString()
+	ctx, cancel := context.WithCancel(context.Background())
+	ack := make(chan bool)
+
+	go func() {
+		for i := 0; i < 10; i++ {
+
+			journal.Send(fmt.Sprintf("test %s - %d", id, i), journal.PriInfo, map[string]string{"TEST_ID": id})
+			select {
+			case <-ack:
+				continue
+			}
+		}
+		cancel()
+	}()
+
+	r, err := NewReader(FormatJSON{}, OptionMatchOR([]JournalEntryMatch{
+		{
+			Field: "TEST_ID",
+			Value: id,
+		},
+	}))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type logEntry struct {
+		Fields map[string]string `json:"fields"`
+	}
+
+	messageCounter := 0
+	for {
+		select {
+		case <- ctx.Done():
+			if messageCounter != 10 {
+				t.Fatalf("expecting to validate 10 log entries. Validated only %d", messageCounter)
+			}
+			return
+		case <-time.After(time.Second):
+			t.Fatal("too much time to read journal")
+		default:
+			buf := new(bytes.Buffer)
+			r.Follow(time.Second, buf)
```

Comment:
You're not checking the `error` that is returned by `r.Follow()` here.

Reply:
If I got it right, we should check that there _is_ an error here shouldn't we?